prime-react/src/main/scala/react/primereact/Tree.scala

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.primereact
+
+import cats.Eq
+import cats.derived.*
+import cats.syntax.all.*
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import lucuma.typed.primereact.components.{Tree => CTree}
+import lucuma.typed.primereact.primereactStrings.checkbox
+import lucuma.typed.primereact.primereactStrings.multiple
+import lucuma.typed.primereact.primereactStrings.single
+import lucuma.typed.primereact.treeTreeMod.TreeNodeTemplateOptions
+import lucuma.typed.primereact.treenodeTreenodeMod.{TreeNode => CTreeNode}
+import org.scalablytyped.runtime.StringDictionary
+import react.common.Css
+import react.common.ReactFnProps
+
+import scalajs.js
+import scalajs.js.JSConverters.*
+
+case class Tree[A](
+  value:         Seq[Tree.Node[A]],
+  nodeTemplate:  (A, TreeNodeTemplateOptions) => VdomNode,
+  expandedKeys:  Map[Tree.Id, Boolean] = Map.empty,
+  onToggle:      js.UndefOr[Map[Tree.Id, Boolean] => Callback] = js.undefined,
+  selectionMode: js.UndefOr[Tree.SelectionMode] = js.undefined,
+  loading:       js.UndefOr[Boolean] = js.undefined
+) extends ReactFnProps[Tree[A]](Tree.component)
+
+object Tree {
+
+  private def component[A] = ScalaFnComponent[Tree[A]] { props =>
+    CTree
+      .value(props.value.map(_.toJsNode).toJSArray)
+      .applyOrNot(props.selectionMode, (c, p) => c.selectionMode(p.value))
+      .nodeTemplate((c, o) =>
+        props
+          .nodeTemplate(c.asInstanceOf[CTreeNode].data.asInstanceOf[Tree.Node[A]].data, o)
+          .rawNode
+      )
+      .applyOrNot(
+        props.expandedKeys,
+        (c, p) => c.expandedKeys(StringDictionary(p.map { case (k, v) => k.value -> v }.toSeq: _*))
+      )
+      .applyOrNot(
+        props.onToggle,
+        (c, p) => c.onToggle(e => p(e.value.toMap))
+      )
+      .applyOrNot(props.loading, _.loading(_))
+
+  }
+
+  enum SelectionMode(val value: single | multiple | checkbox) derives Eq:
+    case Single   extends SelectionMode(single)
+    case Multiple extends SelectionMode(multiple)
+    case Checkbox extends SelectionMode(checkbox)
+
+  opaque type Id = String
+  object Id:
+    inline def apply(value: String): Id                  = value
+    extension (opaqueValue: Id) inline def value: String = opaqueValue
+
+  case class Node[A](
+    id:       Tree.Id,
+    data:     A,
+    label:    js.UndefOr[String] = js.undefined,
+    icon:     Css = Css.Empty,
```

Comment:
Can we make `icon: FontAwesomeIcon | Image | String` as we have in other components?

(Maybe that should be a type alias)

Reply:
I don't think adding anything other than a `String` actually works in the binding. The typescript types are 'any', but everything is just added as a class added to an element, not as an actual element. So `FontAwesomeIcon` (or any `VdomNode`) won't work

Reply:
I haven't looked at the `CTreeNode` generated type, but in the TS definition of `TreeNode` the icon is defined as `icon?: IconType<TreeNode> | undefined;` and `export type IconType<ParentProps> = React.ReactNode | ((options: IconOptions<ParentProps>) => React.ReactNode);` (https://github.com/primefaces/primereact/blob/2f35bcd8c5c8802d984b9f7edc5d94be85939a57/components/lib/utils/utils.d.ts#L151).

In other places where there's an `IconType<?>` we use `icon: FontAwesomeIcon | Image | String` and call its extension method `toPrime` (or `toPrimeWithClass`) to conform to the facade.

Reply:
Oops, yeah you're totally right, I'll add a type alias and use it here as well